content/5_Plat/RURQ.mdx

```diff
@@ -37,15 +37,77 @@ queries.
 
 ### Implementation
 
-<Resources>
-	<Resource
-		source="Benq"
-		title="BITrange"
-		url="https://github.com/bqi343/USACO/blob/master/Implementations/content/data-structures/1D%20Range%20Queries%20(9.2)/BITrange.h"
-	/>
-</Resources>
+<LanguageSection>
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+class fenwick_tree {
+  private:
+	int N;
+	vector<int> t;
+
+  public:
+	fenwick_tree(int sz) {
+		t.resize(sz + 1);
+		N = sz;
+	}
+	void add(int x, int val) {
+		while (x <= N) {
+			t[x] += val;
+			x += x & -x;
+		}
+	}
+	int get_sum(int x) {
+		int sum = 0;
+		while (x > 0) {
+			sum += t[x];
+			x -= x & -x;
+		}
+		return sum;
+	}
+};
+
+void test_case() {
+	int n, q;
+	cin >> n >> q;
+
+	fenwick_tree bit(n), bit2(n);
```

Comment:
perhaps more informative names than `bit` and `bit2` would be nice

Reply:
Done